### PR TITLE
Spancholi/fix osdu writes

### PR DESCRIPTION
**Pull request by spancholi87**

### CODE REVIEW

File: spark-datasource/src/main/scala/com/microsoft/spark/osdu/OSDUDataWriter.scala [hunk 1]
```diff
@@ -19,60 +19,70 @@ package com.microsoft.spark.osdu
...
       // TODO: retry?
       // up to 500
-      val createOrUpdateRecord = storageApi.createOrUpdateRecords(
-        partitionId,
-        true /* skipdups */,
-        "",
-        recordBuffer.asJava)     
+      //TODO - Remove the if condition in future and use OSDUSDK to post data
+      if (useOSDUSDK) {
+        val client = new ApiClient()
```
Comment (eisber) [inline spark-datasource/src/main/scala/com/microsoft/spark/osdu/OSDUDataWriter.scala:72]: I'd avoid recreating the ApiClient as I think this will tear down the https connection. use a lazy object if you want to delay initialization.

File: spark-datasource/src/main/scala/com/microsoft/spark/osdu/OSDUStorageClient.scala [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * limitations under the License.
+ */
+
+package com.microsoft.spark.osdu
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.microsoft.osdu.client.model.StorageRecord
+import scala.collection.mutable.MutableList
+import org.asynchttpclient.DefaultAsyncHttpClient
```
Comment (eisber) [inline spark-datasource/src/main/scala/com/microsoft/spark/osdu/OSDUStorageClient.scala:24]: why did you move to this one? and not the HTTP library that's used by OSDUSDK... there's been some work using shading to make sure there isn't any conflict w/ spark...

**Review by eisber — CHANGES_REQUESTED** [review]

**Review by spancholi87 — COMMENTED** [review]
Updated code

**Review by eisber — APPROVED** [review]
